parser/ndt.go

```diff
@@ -38,20 +36,38 @@ func (n *NDTParser) ParseAndInsert(meta map[string]bigquery.Value, testName stri
 		// Ignoring non-snaplog file.
 		return nil
 	}
+
+	if len(rawSnapLog) > 10*1024*1024 {
+		metrics.TestCount.With(prometheus.Labels{
+			"table": n.TableName(), "type": "oversize"}).Inc()
+		log.Printf("Ignoring oversize snaplog: %d, %s\n",
+			len(rawSnapLog), testName)
+		return nil
+	}
+
+	// Record the file size.
+	metrics.FileSizeHistogram.Observe(float64(len(rawSnapLog)))
+
 	tmpFile, err := ioutil.TempFile(n.tmpDir, "snaplog-")
 	if err != nil {
+		metrics.TestCount.With(prometheus.Labels{
+			"table": n.TableName(), "type": "no-tmp"}).Inc()
```

Comment:
Something feels funny about these labels on `TestCount`.

I think we'll want a three-tuple so that we know whether "s2c" or "c2s" files are seeing more errors or corruptions. e.g. `{table, type, status}`

This doesn't need to block this PR, but maybe a TODO.